[CORL-1596] Move deleted comments into approve/rejected queues

nick-funk opened the pull request:
## What does this PR do?

When a user is deleted, the comments for that user are approved and rejected accordingly depending on whether they have replies or not.

## What changes to the GraphQL/Database Schema does this PR introduce?

None

## How do I test this PR?

- Create a user
- Create some comments for them
- Get some of the comments to be reported
- Ensure that several of the comments have replies from another commenter too
- Use the following mutation or set the account for deletion and delete the user

```
mutation ($userID: ID!, $clientMutationId: String!) {
  deleteUserAccount(input: {
    userID: $userID,
    clientMutationId: $clientMutationId
  }) {
    user {
      id
    }
  }
}
```

- Make sure the deletion mutation returns successfully
- Check the comment streams and queues
- The comments should either be deleted (no replies) or show a tombstone (has replies)
- The comments should be in the rejected queue (no replies) or in the approved queue (has replies)

nick-funk committed:
Put deleted comments in rejected/approved based on child count

Allows us to get the deleted comments out of the
main moderation queues.

nick-funk committed:
Add missing await from deletion moderateComments loop

wyattjoh commented on src/core/server/services/users/delete.ts:
```diff
@@ -101,12 +109,99 @@ async function deleteUserActionCounts(
+  mongo: Db,
+  redis: AugmentedRedis,
+  tenantID: string,
+  filter: any,
```
This can be typed! Add:

```ts
filter: FilterQuery<Comment>,
```

Where `FilterQuery` is from `coral-server/models/helpers`, and `Comment` is the comment model.

wyattjoh reviewed (changes requested):
The rest looks good! Think there's one refinement where we can use proper types for the argument.

nick-funk committed:
use FilterQuery<Comment> instead of any

Nice typings for Mongo queries in TS

wyattjoh commented on src/core/server/services/users/delete.ts:
```diff
@@ -101,12 +109,99 @@ async function deleteUserActionCounts(
+  mongo: Db,
+  redis: AugmentedRedis,
+  tenantID: string,
+  filter: FilterQuery<Comment>,
```
You need to import the `Comment` interface from `coral-server/models/comment`, this is the global `Comment` interface.

nick-funk committed:
Import missing Comment model

It seems like there is another Comment model
that is by default available in TS or Mongo,
this makes sure we use the right one (the one
we wrote for our Coral services)
